src/workdayApiHelpers/searchHelpers.ts

```diff
@@ -0,0 +1,163 @@
+export let sessionSecureToken: string = ""
+import {
+  getTermFromSectionDetailsString,
+  parseSectionDetails,
+} from "../content/utils"
+import { SupplementaryData, ISectionData } from "../content/App/App.types"
+import { defaultColorList } from "../content/Settings/courseColors"
+
+async function buildRequestOptions(code: string, sessionSecureToken?: string) {
+  const urlencoded = new URLSearchParams()
+  urlencoded.append("q", code)
+
+  const headers = new Headers({
+    "Content-Type": "application/x-www-form-urlencoded",
+  })
+
+  if (sessionSecureToken) {
+    urlencoded.append("sessionSecureToken", sessionSecureToken)
+    headers.set("Session-Secure-Token", sessionSecureToken)
+  }
+
+  return {
+    method: "POST",
+    body: urlencoded,
+    redirect: "follow" as RequestRedirect,
+    headers,
+  }
+}
+
+async function fetchSearchData(url: string, requestOptions: RequestInit) {
+  try {
+    const response = await fetch(url, requestOptions)
+    return await response.json()
+  } catch (error) {
+    console.error("Error fetching data:", error)
```

Comment:
We seem to get a lot of errors from workday, where there's an actual error message but returned as XML (for example [this discord message](https://discord.com/channels/1245111270005542962/1245211192440655954/1253930870532997150)) which causes the JSON parse to fail.

it might be out of scope for this PR, but if we could parse the XML error to return the actual error message, that would be great for debugging. alternatively, printing the XML might be a lower-effort solution.

Reply:
I have a whole PR planned for better error handling :)